src/main/java/seedu/address/model/person/Assignment.java

```diff
@@ -0,0 +1,93 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's assignment in the assignment book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidAssignment(String)}
+ */
+public class Assignment {
+
+    public static final String MESSAGE_CONSTRAINTS =
+        "Assignments should only contain alphanumeric characters "
+        + "and spaces, and it should not be blank";
+
+    /*
+     * The first character of the address must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
+
+    public final Name name;
+    private int score;
+    private int maxScore;
+
+    /**
+     * Constructs an assignment with the given name and maximum score.
+     *
+     * @param name The name of the assignment. Must not be null and must satisfy the constraints.
+     * @param maxScore The maximum possible score for the assignment.
+     * @throws NullPointerException If the provided name is null.
+     * @throws IllegalArgumentException If the provided name does not satisfy the constraints.
+     */
+    public Assignment(Name name, int maxScore) {
+        requireNonNull(name);
+        checkArgument(isValidAssignment(name), MESSAGE_CONSTRAINTS);
+        this.name = name;
+        this.maxScore = maxScore;
+    }
+
+    public Assignment(Name name) {
+        this(name, 100);
+    }
+
+    /**
+     * Returns true if a given string is a valid email.
+     */
+    public static boolean isValidAssignment(Name test) {
+        return test.toString().matches(VALIDATION_REGEX);
+    }
+
+    public int getScore() {
```

Comment:
no javadoc?

Reply:
and for the other methods as well

Reply:
These methods are trivial enough, and are not flagged to be requiring javadocs by checkstyle.